Sources/Host/POSIX/File.cpp

```diff
@@ -54,5 +55,28 @@ ErrorCode File::pread(ByteVector &buf, uint64_t &count, uint64_t offset) {
 
   return _lastError = kSuccess;
 }
+
+ErrorCode File::createDirectory(std::string const &path, uint32_t flags) {
+  size_t pos = 0;
+
+  // Each parent directory must be created individually, if it does not exist
+  while ((pos = path.find('/', pos + 1)) != std::string::npos) {
```

Comment:
Yeah the initialization not being required is me not being able to read
code, ignore that comment.

On Wed, Jun 15, 2016 at 4:02 PM, Francis Ricci <notifications@github.com>
wrote:

> In Sources/Host/POSIX/File.cpp
> <https://github.com/facebook/ds2/pull/375#discussion_r67261680>:
>
> > @@ -54,5 +55,28 @@ ErrorCode File::pread(ByteVector &buf, uint64_t &count, uint64_t offset) {
> >
> >    return _lastError = kSuccess;
> >  }
> > +
> > +ErrorCode File::createDirectory(std::string const &path, uint32_t flags) {
> > +  size_t pos = 0;
> > +
> > +  // Each parent directory must be created individually, if it does not exist
> > +  while ((pos = path.find('/', pos + 1)) != std::string::npos) {
>
> Why is the initialization not required? You have to tell find to start at
> the beginning of the string.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/ds2/pull/375/files/687555e04395f15d83108877ce62d7c6805a3aa9..29a134830a0744f8aad6e24930060c19ebd6684f#r67261680>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AADHbMsA9AkWwVpwxetelkTrhdXjK6Lkks5qMISEgaJpZM4I229L>
> .
>



-- 
Stephane Sezer